packages/office-addin-manifest/src/manifestHandler/manifestHandlerJson.ts

```diff
@@ -1,29 +1,69 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
+import * as fs from "fs-extra";
+import { TeamsAppManifest } from "@microsoft/teamsfx-api";
+import { v4 as uuidv4 } from "uuid";
 import { ManifestInfo } from "../manifestInfo";
 import { ManifestHandler } from "./manifestHandler";
 
 export class ManifestHandlerJson extends ManifestHandler {
   /* eslint-disable @typescript-eslint/no-unused-vars */
-  async modifyManifest(guid?: string, displayName?: string): Promise<any> {
-    throw new Error("Manifest cannot be modified in .json files");
+  async modifyManifest(guid?: string, displayName?: string): Promise<TeamsAppManifest> {
+    const teamsAppManifest: TeamsAppManifest = await JSON.parse(this.fileData);
+
+    if (typeof guid !== "undefined") {
+      if (!guid || guid === "random") {
+        guid = uuidv4();
+      }
+      teamsAppManifest.id = guid;
+    }
+
+    if (typeof displayName !== "undefined") {
+      teamsAppManifest.name.short = displayName;
+    }
+    return teamsAppManifest;
   }
 
   async parseManifest(): Promise<ManifestInfo> {
-    const manifestInfo: ManifestInfo = new ManifestInfo();
-
     try {
-      const file = await JSON.parse(this.fileData);
+      const file: TeamsAppManifest = await JSON.parse(this.fileData);
+      return this.translateTeamsAppManifestToManifestInfo(file);
     } catch (err) {
       throw new Error(`Unable to read data for manifest file: ${this.manifestPath}. \n${err}`);
     }
+  }
+
+  translateTeamsAppManifestToManifestInfo(teamsAppManifest: TeamsAppManifest): ManifestInfo {
```

Comment:
Nit: Shorter to the point name might be a little better.  Like "getManifestInfo" or something to that effect.